content/docs/iac/guides/migration/aws-import-ids.md

```diff
@@ -0,0 +1,208 @@
+---
+title_tag: "Finding AWS import IDs and special cases"
+meta_desc: Learn how to find AWS resource IDs and handle CloudFormation-specific edge cases when importing resources into Pulumi.
+title: AWS import IDs and special cases
+h1: "Finding AWS import IDs and special cases"
+meta_image: /images/docs/meta-images/docs-meta.png
+aliases:
+    - /docs/iac/guides/migration/migrating-to-pulumi/aws-import-ids/
+menu:
+    iac:
+        name: Finding AWS import IDs
+        parent: iac-guides-migration
+        weight: 3
+---
+
+This guide explains how to discover the correct AWS resource IDs to use when importing existing CloudFormation- or CDK-managed resources into Pulumi, and calls out CloudFormation features that require special handling.
+
+## Finding Resource IDs
+
+Every import requires a resource ID. Whether you are running `pulumi import` for a single resource or using a bulk `import.json` file (for example, generated with `pulumi preview --import-file import.json`), you need to supply an ID value that uniquely identifies the existing AWS resource. Sometimes this is a single value like a resource `ARN`, but other times it is a composite id made up of a combination of property values. For example, to import a Lambda Permission resource you need the `functionName` and the Permission `id` and it needs to be in the format `functionName|id`.
+
+The first place you should start is the ids from the CloudFormation stack. Most of the time the values for the import ids can be extracted from the `PhysicalResourceId` in the CloudFormation stack.
+
+```bash
+$ aws cloudformation list-stack-resources --stack-name my-stack \
+    --query 'StackResourceSummaries[].{Type:ResourceType,LogicalResourceId:LogicalResourceId,Physical:PhysicalResourceId}'
+```
+
+Optionally, you can use the `cdk2pulumi` tool to lookup information on the import ids. It has an `ids` subcommand that returns information on the expected import id format. For example:
+
+```bash
+$ pulumi plugin run cdk2pulumi -- ids AWS::Lambda::Permission
+Resource: aws-native:lambda:Permission (CFN: AWS::Lambda::Permission, provider: aws-native)
+Format: {functionName}|{id}
+Parts:
+  - functionName (Input): The name or ARN of the Lambda function, version, or alias.
+  **Name formats**
+ +  *Function name* – ``my-function`` (name-only), ``my-function:v1`` (with alias).
+  +  *Function ARN* – ``arn:aws:lambda:us-west-2:123456789012:function:my-function``.
+  +  *Partial ARN* – ``123456789012:function:my-function``.
+
+ You can append a version number or alias to any of the formats. The length constraint applies only to the full ARN. If you specify only the function name, it is limited to 64 characters in length.
+  - id (Output)
+```
+
+### Finding IDs example
+
+Lets walk through an example of finding the import ids for a couple of resources.
+
+**1. List the stack resources**
```

Comment:
These should just be standard ordered lists, e.g.:

```markdown
1. List the stack resources:

    ```bash
    do stuff
    ```

1. (do the next thing)
```

Indent code fences 4 spaces and keep a blank line in between and they will render correctly.